src/core/layouts/index.ts

```diff
@@ -5,11 +5,104 @@
 import { html } from 'lighterhtml'
 import { CSSProperties } from 'react'
 import { LayoutDeclaration, Transition } from '../../compositor/layouts'
+import { getProject } from '../data'
+import CoreContext from '../context'
 
 type LayoutFreeProps = {
   size: { x: string; y: string }
   position: { x: string; y: string }
   opacity: number
+  type: 'alert'
+  preset: string
+}
+
+const getPresetStyle = (preset: string) => {
```

Comment:
Only question is whether it's worth using an enum or union type here. I would recommend at least adding a doc comment to the `LayoutFreeProps.preset` property, since it's not immediately clear what "preset" means without reading deeper into the code.